pkg/api/vm/handler_test.go

```diff
@@ -622,3 +644,239 @@ func (c fakeVirtualMachineInstanceMigrationClient) Watch(namespace string, opts
 func (c fakeVirtualMachineInstanceMigrationClient) Patch(namespace, name string, pt types.PatchType, data []byte, subresources ...string) (result *kubevirtv1.VirtualMachineInstanceMigration, err error) {
 	return c(namespace).Patch(context.TODO(), name, pt, data, metav1.PatchOptions{}, subresources...)
 }
+
+type fakeNodeCache func() corev1type.NodeInterface
+
+func (c fakeNodeCache) Get(name string) (*corev1.Node, error) {
+	return c().Get(context.TODO(), name, metav1.GetOptions{})
+}
+
+func (c fakeNodeCache) List(selector labels.Selector) ([]*corev1.Node, error) {
+	list, err := c().List(context.TODO(), metav1.ListOptions{LabelSelector: selector.String()})
+	if err != nil {
+		return nil, err
+	}
+	result := make([]*corev1.Node, 0, len(list.Items))
+	for i := range list.Items {
+		result = append(result, &list.Items[i])
+	}
+	return result, err
+}
+
+func (c fakeNodeCache) AddIndexer(indexName string, indexer ctlcorev1.NodeIndexer) {
+	panic("implement me")
+}
+
+func (c fakeNodeCache) GetByIndex(indexName, key string) ([]*corev1.Node, error) {
+	panic("implement me")
+}
```

Comment:
resue the fakeclient on the https://github.com/harvester/harvester/tree/master/pkg/util/fakeclients